src/baconian.rs

```diff
@@ -0,0 +1,467 @@
+lazy_static! {
+    static ref ITALIC_CODES: HashMap<&'static str, char> = hashmap!{
+        // Using Mathematical Italic
+        "A" => '\u{1D434}',
+        "B" => '\u{1D435}',
+        "C" => '\u{1D436}',
+        "D" => '\u{1D437}',
+        "E" => '\u{1D438}',
+        "F" => '\u{1D439}',
+        "G" => '\u{1D43a}',
+        "H" => '\u{1D43b}',
+        "I" => '\u{1D43c}',
+        "J" => '\u{1D43d}',
+        "K" => '\u{1D43e}',
+        "L" => '\u{1D43f}',
+        "M" => '\u{1D440}',
+        "N" => '\u{1D441}',
+        "O" => '\u{1D442}',
+        "P" => '\u{1D443}',
+        "Q" => '\u{1D444}',
+        "R" => '\u{1D445}',
+        "S" => '\u{1D446}',
+        "T" => '\u{1D447}',
+        "U" => '\u{1D448}',
+        "V" => '\u{1D449}',
+        "W" => '\u{1D44a}',
+        "X" => '\u{1D44b}',
+        "Y" => '\u{1D44c}',
+        "Z" => '\u{1D44d}',
+        // Using Mathematical Sans-Serif Italic
+        "a" => '\u{1D622}',
+        "b" => '\u{1D623}',
+        "c" => '\u{1D624}',
+        "d" => '\u{1D625}',
+        "e" => '\u{1D626}',
+        "f" => '\u{1D627}',
+        "g" => '\u{1D628}',
+        "h" => '\u{1D629}',
+        "i" => '\u{1D62a}',
+        "j" => '\u{1D62b}',
+        "k" => '\u{1D62c}',
+        "l" => '\u{1D62d}',
+        "m" => '\u{1D62e}',
+        "n" => '\u{1D62f}',
+        "o" => '\u{1D630}',
+        "p" => '\u{1D631}',
+        "q" => '\u{1D632}',
+        "r" => '\u{1D633}',
+        "s" => '\u{1D634}',
+        "t" => '\u{1D635}',
+        "u" => '\u{1D636}',
+        "v" => '\u{1D637}',
+        "w" => '\u{1D638}',
+        "x" => '\u{1D639}',
+        "y" => '\u{1D63a}',
+        "z" => '\u{1D63b}'
+    };
+}
+
+/// Get the code for a given key (source character)
+fn get_code(distinct: bool, key: &str) -> String {
+    let mut code = String::new();
+    if distinct {
+        if DISTINCT_CODES.contains_key(key.to_uppercase().as_str()) {
+            code.push_str(DISTINCT_CODES.get(key.to_uppercase().as_str()).unwrap());
+        }
+    } else {
+        // Need to handle 'I'/'J' and 'U'/'V'
+        let mut key_upper = key.to_uppercase();
+
+        match key_upper.as_str() {
+            "J" => key_upper = "I".to_string(),
+            "U" => key_upper = "V".to_string(),
+            _ => {}
+        }
+        if TRAD_CODES.contains_key(key_upper.as_str()) {
+            code.push_str(TRAD_CODES.get(key_upper.as_str()).unwrap());
+        }
+    }
+
+    code
+}
+
+/// Gets the key (the source character) for a given cipher text code
+fn get_key(distinct: bool, code: &str) -> String {
+    let mut key = String::new();
+
+    let codes = if distinct {
+        DISTINCT_CODES.iter()
+    } else {
+        TRAD_CODES.iter()
+    };
+    for (_key, val) in codes {
+        if val == &code {
+            key.push_str(_key);
+        }
+    }
+    key
+}
+
+/// This struct is created by the `new()` method. See its documentation for more.
+pub struct Baconian {
+    distinct: bool,
```

Comment:
Could you please explain to me the purpose of the `distinct`. This variable name does not seem intuitive to me...

Reply:
How about `non_traditional_dictionary`? I'd like to keep the default behaviour as `traditional` as that is how the original cipher was used.

Reply:
How about: `use_distinct_alphabet`?

Reply:
Okay will sort this. Rammed busy right now, but will get on to it this week.